lib/git.js

```diff
@@ -236,6 +237,29 @@ class Git extends EventEmitter {
           });
       }
   }
+  /**
+   * Authenticate the user with a basic authentication
+   * @method authenticate
```

Comment:
can we call the method authenticate?

Reply:
We can rename it, of course. But I think "authenticate" won't work (if I remember, I tried before and didn't work) because there is also a variable with that name.

Reply:
Let me know which name would you want and I'll commit it in a second.

Reply:
Maybe 'login'?

Reply:
oh in the case that the user can override login it should be `_authenticate`. Login would infer that there is a session, in the case of git, there is a constant affirmation that the user is who they say they are.

Reply:
Good thinking.